kernel/src/hil/symmetric_encryption.rs

```diff
@@ -1,39 +1,131 @@
-//! Interfaces for accessing encryption and decryption of symmetric ciphers.
+//! Interface for symmetric-cipher encryption
 //!
-//! Only AES-128-ctr supported at the moment.
-//!
-//! The interface is supposed to work for hardware supported crypto but should
-//! work for software implemented crypto as well.
-//!
-//! State Machine:
-//!
-//! 1. `init()`
-//! 2. `set_key()`
-//! 3. `aes128_crypt_ctr()`: can be used arbitrary number of times.
+//! see boards/imix/src/aes_test.rs for example usage
 
 use returncode::ReturnCode;
 
-pub trait SymmetricEncryption {
-    /// Set the callback client.
-    fn set_client(&self, client: &'static Client);
+/// Implement this trait and use `set_client()` in order to receive callbacks from an `AES128`
+/// instance.
+pub trait Client<'a> {
```

Comment:
Seems like this should be "AESClient" if the related trait defined below is called "CCMClient". Just "Client" seems a little vague